app/ante/eth.go

```diff
@@ -213,6 +215,11 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		gasPool.ConsumeGas(ctx.GasMeter().GasConsumedToLimit(), "gas pool check")
 	}
 
+	// Set newCtx.GasMeter to 0, with a limit of GasWanted (gasLimit)
+	if ctx.IsCheckTx() {
```

Comment:
Sorry for the confusion from my comments, but we should do it for deliverTx too, but not for msg handler, so we should remove this condition, but override with a infinite gas meter at the beginning of `Keeper.EthereumTx` method.

Reply:
@yihuang , here:
https://github.com/tharsis/ethermint/blob/caa1c5a6c6b7ed8ba4aaf6e0b0848f6be5ba6668/x/evm/keeper/msg_server.go#L24-L26, right?

Reply:
yes